src/app/users.component.html

```diff
@@ -2,45 +2,38 @@ <h2>Users</h2>
 <div class="toolbar">
   <mat-form-field class="search">
     <mat-icon matPrefix>search</mat-icon>
-    <input #in (input)="data.queryBy({search: in.value})" type="text" matInput placeholder="Search">
+    <label>
+      <input #input (input)="data.queryBy({search: input.value})" type="text" matInput placeholder="Search">
+    </label>
   </mat-form-field>
   <mat-form-field class="registration">
-    <input (dateChange)="data.queryBy({registration: $event.value})" matInput [matDatepicker]="picker" placeholder="Registration"/>
+    <label>
+      <input (dateChange)="data.queryBy({registration: $event.value})" matInput [matDatepicker]="picker" placeholder="Registration"/>
+    </label>
     <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
     <mat-datepicker #picker></mat-datepicker>
   </mat-form-field>
-  <mat-form-field class="property">
-    <mat-label>Order by</mat-label>
-    <mat-select value="username" (selectionChange)="data.sortBy({property: $event.value})">
-      <mat-option value="id">ID</mat-option>
-      <mat-option value="username">Username</mat-option>
-    </mat-select>
-  </mat-form-field>
-  <mat-button-toggle-group value="asc" (change)="data.sortBy({order: $event.value})">
-    <mat-button-toggle value="asc"><mat-icon>arrow_upward</mat-icon></mat-button-toggle>
-    <mat-button-toggle value="desc"><mat-icon>arrow_downward</mat-icon></mat-button-toggle>
-  </mat-button-toggle-group>
   <mat-spinner *ngIf="data.loading$ | async" diameter="32"></mat-spinner>
 </div>
-<table mat-table [dataSource]="data" class="mat-elevation-z1">
+<table mat-table [dataSource]="data" matSort matSortActive="username" matSortStart="asc" matSortDisableClear class="mat-elevation-z1">
 
    <ng-container matColumnDef="id">
-    <th mat-header-cell *matHeaderCellDef> ID </th>
+    <th mat-header-cell *matHeaderCellDef mat-sort-header> ID </th>
     <td mat-cell *matCellDef="let user"> {{user.id}} </td>
   </ng-container>
 
   <ng-container matColumnDef="username">
-    <th mat-header-cell *matHeaderCellDef> Username </th>
+    <th mat-header-cell *matHeaderCellDef mat-sort-header> Username </th>
     <td mat-cell *matCellDef="let user"> {{user.username}} </td>
   </ng-container>
 
   <ng-container matColumnDef="email">
-    <th mat-header-cell *matHeaderCellDef> Mail </th>
+    <th mat-header-cell *matHeaderCellDef mat-sort-header> Mail </th>
     <td mat-cell *matCellDef="let user"> {{user.email}} </td>
   </ng-container>
 
   <ng-container matColumnDef="registration">
-    <th mat-header-cell *matHeaderCellDef> Registration </th>
+    <th mat-header-cell *matHeaderCellDef mat-sort-header> Registration </th>
```

Comment:
Without providing an ID for `mat-sort-header` it'll use the column-def name. Here that's "registration", however, the sorting property is "registrationDate". So we either need to change the column-def or provide the proper ID to `mat-sort-header`.